tests/common/member_test.go

```diff
@@ -52,3 +55,157 @@ func TestMemberList(t *testing.T) {
 		})
 	}
 }
+
+func TestMemberAdd(t *testing.T) {
+	testRunner.BeforeTest(t)
+
+	tcs := []struct {
+		name    string
+		learner bool
+	}{
+		{
+			name:    "NotLearner",
+			learner: false,
+		},
+		{
+			name:    "Learner",
+			learner: true,
+		},
+	}
+
+	for _, tc := range tcs {
+		for _, nc := range clusterTestCases {
+			t.Run(tc.name+"/"+nc.name, func(t *testing.T) {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				clus := testRunner.NewCluster(ctx, t, nc.config)
+				defer clus.Close()
+				cc := clus.Client()
+
+				testutils.ExecuteUntil(ctx, t, func() {
+					clus.WaitLeader(t)
+					var addResp *clientv3.MemberAddResponse
+					var err error
+					if tc.learner {
+						addResp, err = cc.MemberAddAsLearner(ctx, "newmember", []string{"http://localhost:123"})
+					} else {
+						addResp, err = cc.MemberAdd(ctx, "newmember", []string{"http://localhost:123"})
+					}
+					if nc.config.ClusterSize == 1 {
+						checkAddResp(t, addResp, err)
+					} else {
+						if err == nil {
+							t.Fatalf("expect MemberAdd to fail, got no error, addResp: %v", addResp)
+						}
+						if !strings.Contains(err.Error(), "etcdserver: unhealthy cluster") {
+							t.Fatalf("expected: 'etcdserver: unhealthy cluster' error,  got: %s", err)
+						}
+					}
+				})
+			})
+		}
+	}
+}
+
+func TestMemberAdd_SleetHealthInterval(t *testing.T) {
```

Comment:
I don't see similar test removed in this PR. Where this test has come from?

Reply:
This is a scenario not covered before: 

- In the first 5 seconds(HealthInterval) after the creation of a cluster(with `--strict-reconfig-check` enabled), any call to `MemberAdd` and `MemberRemove` should return an `etcdserver: unhealthy cluster` error. 
- As the first 5 seconds(HealthInterval) has passed, call to these functions should be successful.

This behaviour is controlled by the `isConnectedFullySince` function:
https://github.com/etcd-io/etcd/blob/d05c894e854e1b8545c41ff26ac85c8b76917a7c/server/etcdserver/server.go#L1350-L1358

Reply:
ok, please rename/rewrite test so the reason for the behavior is known. For example `TestMemberAdd_BeforeConnectedToAllPeers`.